queue.go

```diff
@@ -103,3 +103,44 @@ func (w *WaitQueue) Unlock() {
 		w.work.Unlock()
 	}
 }
+
+// BoundedMutuallyExclusiveWaitQueue is a work queue that allows a bounded number of tasks to execute
+// at any one time, but only one of a particular task type
+type BoundedMutuallyExclusiveWaitQueue struct {
+	sem    chan int
+	queues map[string]*WaitQueue
+}
+
+// New returns a new BoundedMutuallyExclusiveWaitQueue ready for use.
+func NewBounded(size int, queueName string, additionalQueueNames ...string) *BoundedMutuallyExclusiveWaitQueue {
```

Comment:
Is golang's compiler smart enough to flag zero length variadic args + len check as a compile error if there are any calls to `NewBounded` that do not specify any queue names? I was under the assumption that it wasn't, so I wanted to ensure that we'd get a compile error in the aforementioned case rather than a runtime error.